classes/manager.php

```diff
@@ -80,12 +80,17 @@ public function get_starttime(): float {
     /**
      * Get manager instance.
      *
+     * @param bool $initialise If true, will initialise the manager upon creation.
+     *
      * @return manager
      * @throws \dml_exception
      */
-    public static function get_instance(): manager {
+    public static function get_instance(bool $initialise = true): manager {
         if (!self::$instance) {
             self::create();
+            if ($initialise) {
+                self::$instance->init();
+            }
```

Comment:
Nothing appears to call get_instance with $initialise = false, could we just always init it then?